### PR TITLE
Add flake8-bugbear code B007

### PR DESCRIPTION
**Pull request by allispaul**
Fixes flake8-bugbear code B007, from #106571

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @voznesenskym @penguinwu @EikanWang @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @ngimel @anijain2305

### CODE REVIEW

File: test/quantization/fx/test_quantize_fx.py [hunk 1]
```diff
@@ -5566,11 +5566,13 @@ def forward(self, x):
...
-        options = itertools.product([M1, M2], [True, False])
-        for M, is_qat in options:
+        for is_qat in [True, False]:
             m = M1().eval()
             example_inputs = (torch.randn(1, 3, 3, 3),)
-            m = prepare_fx(m, get_default_qconfig_mapping(), example_inputs=example_inputs)
+            if is_qat:
+                m = prepare_qat_fx(m, get_default_qat_qconfig_mapping(), example_inputs=example_inputs)
+            else:
+                m = prepare_fx(m, get_default_qconfig_mapping(), example_inputs=example_inputs)
```
Comment (thiagocrepaldi) [inline test/quantization/fx/test_quantize_fx.py:5575]: Is this related to flake8-bugbear? seems quantization-specific changes, not a lint

File: test/quantization/fx/test_quantize_fx.py [hunk 2]
```diff
@@ -5583,11 +5585,14 @@ def forward(self, x):
                 expected_node_list=node_list)
 
             m = M2().eval()
-            m = prepare_fx(m, get_default_qconfig_mapping(), example_inputs=example_inputs)
+            if is_qat:
+                m = prepare_qat_fx(m, get_default_qat_qconfig_mapping(), example_inputs=example_inputs)
+            else:
+                m = prepare_fx(m, get_default_qconfig_mapping(), example_inputs=example_inputs)
```
Comment (thiagocrepaldi) [inline test/quantization/fx/test_quantize_fx.py:5591]: ditto

**Review by thiagocrepaldi — COMMENTED** [review]
A few changes don't seem to be a lint fix, but actual changes related to qat. 
Please check whether they are intentional

File: test/quantization/fx/test_quantize_fx.py (hunk 1 above)
Comment (allispaul) [reply test/quantization/fx/test_quantize_fx.py:5575]: Right, the original version of this test had an unused parameter `is_qat`, and this was my (possibly incorrect) attempt to guess what it was for. I can just remove the parameter if that would be better.

Comment (albanD) [reply test/quantization/fx/test_quantize_fx.py:5575]: We should not change any behavior in these linting PRs. 
In this case, you either want to remove the argument if BC-compatible. Otherwise silence and open an issue for the specific maintainer to fix it.

**Review by albanD — CHANGES_REQUESTED** [review]
I think the best way to merge this PR would be to ensure we just silence anything that requires a behavior change and open issues for them.
Again, we should not change any behavior in such PRs.

File: test/quantization/fx/test_quantize_fx.py (hunk 1 above)
Comment (allispaul) [reply test/quantization/fx/test_quantize_fx.py:5575]: OK, got it!

File: functorch/dim/reference.py [hunk 3]
```diff
@@ -492,7 +492,7 @@ def add_dims(t):
                 dims_seen.record(idx)
                 view_sizes.append(sz)
             elif isinstance(idx, (tuple, list)) and idx and isinstance(idx[0], Dim):
-                for d in idx:
+                for d in idx:  # noqa: B007
```
Comment (albanD) [inline functorch/dim/reference.py:495]: cc @zdevito FYI

File: test/functorch/test_ops.py [hunk 4]
```diff
@@ -1143,7 +1143,7 @@ def test():
                 args = tuple(arg_values) + tuple(kwarg_values)
                 fn, args = get_jvp_variant_primals_tangents(op, sample)
                 is_batch_norm_and_training = is_batch_norm_training(op.name, kwarg_values)
-                for loop_out, batched_out in get_fallback_and_vmap_exhaustive(
+                for _loop_out, _batched_out in get_fallback_and_vmap_exhaustive(
```
Comment (albanD) [inline test/functorch/test_ops.py:1146]: These 3 changes don't look right?

File: test/nn/test_convolution.py [hunk 5]
```diff
@@ -248,11 +248,11 @@ def test_Conv3d_module_same_padding(self):
 
     @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
     def test_thnn_conv_strided_padded_dilated(self):
-        for convfn, dims, transposed in (
-                (torch.nn.functional.conv2d, 2, False),
-                (torch.nn.functional.conv_transpose2d, 2, True),
-                (torch.nn.functional.conv3d, 3, False),
-                (torch.nn.functional.conv_transpose3d, 3, True)):
+        for convfn, dims in (
```
Comment (albanD) [inline test/nn/test_convolution.py:251]: FYI @mikaylagawarecki this might be a bug!

File: test/quantization/core/experimental/apot_fx_graph_mode_ptq.py [hunk 6]
```diff
@@ -31,7 +31,7 @@
 def calibrate(model, data_loader):
     model.eval()
     with torch.no_grad():
-        for image, target in data_loader:
+        for image, _target in data_loader:
```
Comment (albanD) [inline test/quantization/core/experimental/apot_fx_graph_mode_ptq.py:34]: Should be _ ?

File: test/quantization/core/test_quantized_module.py [hunk 7]
```diff
@@ -201,7 +201,7 @@ def _test_linear_api_impl(self, qlinear_module, module_name, qlinear_op,
         self.assertEqual(qlinear.scale, loaded_from_package.scale)
         self.assertEqual(qlinear.zero_point, loaded_from_package.zero_point)
 
-        for name, module in loaded_from_package.named_modules():
+        for name, _module in loaded_from_package.named_modules():
```
Comment (albanD) [inline test/quantization/core/test_quantized_module.py:204]: Should be  _

File: test/quantization/core/test_quantized_tensor.py [hunk 8]
```diff
@@ -1052,8 +1052,8 @@ def _test_qtensor_masked_fill(self, device):
         mask = torch.randint(0, 2, (numel, ), device=device)
         mask = mask.bool()
         x = torch.rand(numel, device=device)
-        qx = torch.quantize_per_tensor(x, scale=scale, zero_point=zero_point, dtype=qtype)
         for qtype, fill_with in itertools.product(types, fills):
+            qx = torch.quantize_per_tensor(x, scale=scale, zero_point=zero_point, dtype=qtype)
```
Comment (albanD) [inline test/quantization/core/test_quantized_tensor.py:1056]: Behavior change, please revert that.

File: test/quantization/core/test_workflow_ops.py [hunk 9]
```diff
@@ -334,8 +334,8 @@ def _test_forward_per_tensor_cachemask_impl(self, device):
         float_types = (torch.float32, torch.float16, torch.float64)
         torch_types = (torch.qint8, torch.quint8)
         Xs = (torch.randn(4, 8, device=device), torch.randn(4, 16, device=device)[:, ::2])
-        tensor_qparam = (True, False)
-        for float_type, torch_type, X, tensor_qparams in itertools.product(float_types, torch_types, Xs, tensor_qparam):
+        tensor_qparams = (True, False)
```
Comment (albanD) [inline test/quantization/core/test_workflow_ops.py:337]: Behavior change please revert
cc @vkuzo this is most likely a bug

File: test/quantization/fx/test_model_report_fx.py [hunk 10]
```diff
@@ -1322,7 +1322,7 @@ def test_input_weight_equalization_determine_points(self):
             fused = self._get_prepped_for_calibration_model(self.TwoBlockComplexNet(), detector_set, fused=True)
 
             # reporter should still give same counts even for fused model
-            for prepared_for_callibrate_model, mod_report in [non_fused, fused]:
+            for prepared_for_callibrate_model, _mod_report in [non_fused, fused]:
```
Comment (albanD) [inline test/quantization/fx/test_model_report_fx.py:1325]: _ ?

File: test/quantization/fx/test_quantize_fx.py [hunk 11]
```diff
@@ -1778,7 +1778,7 @@ def forward(self, x):
                  None),
             ]
             for (ModuleClass, module_constructor_inputs,
-                 inputs, quantized_node, weight_prepack_node) in tests:
+                 inputs, _quantized_node, weight_prepack_node) in tests:
```
Comment (albanD) [inline test/quantization/fx/test_quantize_fx.py:1781]: _ ?

File: test/quantization/fx/test_quantize_fx.py [hunk 12]
```diff
@@ -5566,8 +5566,7 @@ def forward(self, x):
                 x = x.reshape()
                 return x
 
-        options = itertools.product([M1, M2], [True, False])
-        for M, is_qat in options:
+        for is_qat in [True, False]:  # noqa: B007
```
Comment (albanD) [inline test/quantization/fx/test_quantize_fx.py:5569]: FYI @vkuzo this is most likley a bug

**Review by albanD — CHANGES_REQUESTED** [review]
Ok, I only got midway through but there are too many behavior change sneaking in this PR. It has to be broken down to be able to be landed so that it can be properly reviewed.

Splitted by file location / theme should be simplest, I'm happy to find the right reviewers for each if you ping me there.

**Comment by allispaul** [comment]
@albanD Thanks for taking a look at it. You meant that I should withdraw this PR and split it into several smaller ones?

File: test/quantization/fx/test_quantize_fx.py (hunk 11 above)
Comment (allispaul) [reply test/quantization/fx/test_quantize_fx.py:1781]: To explain, the linter is OK with unaccessed variables in for loops if they start with an underscore. In a few cases it seemed worth it to keep the variable name, if the variable was something non-obvious, to make the code more readable. But I can change these to just _ if that would be better.

File: test/functorch/test_ops.py (hunk 4 above)
Comment (allispaul) [reply test/functorch/test_ops.py:1146]: What do you mean?

**Comment by albanD** [comment]
That would be great yes. I think that otherwise, there is too high a risk of a small subtle bug sneaking in.

**Comment by allispaul** [comment]
@albanD I'm still finding my way around this project and this PR touches a lot of stuff. Does this sound like a reasonable breakdown?

distributed
quantization
jit (each ~25 files)
everything else in torch/testing and test/ (~70 files)
everything else (~60 files)

**Comment by albanD** [comment]
That sounds like a good split yes. Thanks!
